enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyAuthenticator.java

```diff
@@ -71,17 +72,21 @@ private void dropAPIKeyDataFromAPIKeyHeader(RequestContext requestContext) throw
         JSONObject jsonObject = getDecodedAPIKeyData(apiKeyHeaderValue);
         jsonObject.remove(APIKeyConstants.API_KEY_JSON_KEY);
         // Update the header with the new API key data.
-        String encodedKeyData = Base64.getEncoder().encodeToString(jsonObject.toJSONString().getBytes());
-        String newAPIKeyHeaderValue = APIKeyConstants.API_KEY_PREFIX + encodedKeyData + checksum;
-        // Add the new header.
-        requestContext.addOrModifyHeaders(ConfigHolder.getInstance().getConfig().getApiKeyConfig()
-                .getApiKeyInternalHeader().toLowerCase(), newAPIKeyHeaderValue);
+        if (!jsonObject.isEmpty()) {
+            String encodedKeyData = Base64.getEncoder().encodeToString(jsonObject.toJSONString().getBytes());
+            String newAPIKeyHeaderValue = APIKeyConstants.API_KEY_PREFIX + encodedKeyData + checksum;
+            // Add the new header.
+            requestContext.addOrModifyHeaders(APIKeyConstants.INTERNAL_API_KEY_HEADER, newAPIKeyHeaderValue);
+        }
     }
 
     private String getAPIKeyFromRequest(RequestContext requestContext) {
         Map<String, String> headers = requestContext.getHeaders();
-        return headers.get(ConfigHolder.getInstance().getConfig().getApiKeyConfig()
-                .getApiKeyInternalHeader().toLowerCase());
+        return ConfigHolder.getInstance().getConfig().getApiKeyConfig().getApiKeyInternalHeaders().stream()
```

Comment:
I guess order is preserved here, If so we should be fine.

Reply:
Yes. Configurations are added in the order, but there's no real difference between the order of checking. Request will be authenticated with either `choreo-api-key` or `api-key` header.

Reply:
what If someone is using both headers but different values, could that bring an issue?

Reply:
Both of these headers are reserved for the API-Key functionality, hence it should be fine to consider that these headers won't be used for other purposes.